### PR TITLE
Port existing networking content where possible

### LINKED ISSUE #30397: Port, redirect and modernize all .NET networking content.
Port, redirect and modernize all .NET networking content. There is a whole sub-doc-set located here:

- [Network Programming in the .NET Framework](https://docs.microsoft.com/dotnet/framework/network-programming)

Much of this content is severely out-of-date, and many code examples reference obsolete APIs. The strategy is to move (port) conceptual content from the **Network Programming in the .NET Framework** under the modern **.NET Fundamentals / Runtime Libraries / Networking** TOC:

image

We should converge the existing HTTP-specific content under a new TOC node, **.NET Fundamentals / Runtime Libraries / Networking / HTTP**. Then we should create several new nodes to capture content specific to additional protocols, such as TCP, UDP, FTP, WebSockets, and whatever else I might be missing. A lot of the legacy content specific to .NET Framework can be modernized and code samples updated accordingly. 

Asynchronous patterns such as:

- `Begin{X}` / `End{X}`
- `AsyncCallback`
- `IAsyncResult`

Should be removed in favor of, `async/await`, `Task`, and `ValueTask` (or other task-like alternatives). conceptual rewrites should strive to serve new learned audiences, making adoption easy.

/cc @richlander

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Port existing networking content where possible:

- Modernize code examples
- Edit pass
- Restructure content
- Remove obsolete bits

Fixes #30397

## Internal preview

- [Network programming in .NET](https://review.docs.microsoft.com/en-us/dotnet/fundamentals/networking/overview?branch=pr-en-us-30759)
- [Network availability](https://review.docs.microsoft.com/en-us/dotnet/fundamentals/networking/network-info?branch=pr-en-us-30759)
- [IPv6 overview](https://review.docs.microsoft.com/en-us/dotnet/fundamentals/networking/ipv6-overview?branch=pr-en-us-30759)
- [HTTP support in .NET](https://review.docs.microsoft.com/en-us/dotnet/fundamentals/networking/http/http-overview?branch=pr-en-us-30759)
- [Make HTTP requests with the `HttpClient` class](https://review.docs.microsoft.com/en-us/dotnet/fundamentals/networking/http/httpclient?branch=pr-en-us-30759)
- [Sockets in .NET](https://review.docs.microsoft.com/en-us/dotnet/fundamentals/networking/sockets/sockets-overview?branch=pr-en-us-30759)
- [Use Socket to send and receive data](https://review.docs.microsoft.com/en-us/dotnet/fundamentals/networking/sockets/socket-services?branch=pr-en-us-30759)
- [TCP in .NET](https://review.docs.microsoft.com/en-us/dotnet/fundamentals/networking/tcp/tcp-overview?branch=pr-en-us-30759)
- [Use TcpClient and TcpListener](https://review.docs.microsoft.com/en-us/dotnet/fundamentals/networking/tcp/tcp-services?branch=pr-en-us-30759)

## Updated TOCs

**_New TOC under .NET fundamentals / Runtime libraries / Networking_**
image

**_Refined TOC under .NET Framework / Network Programming_**
image

### CODE REVIEW

File: docs/framework/network-programming/best-practices-for-system-net-classes.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ The following recommendations will help you use the classes contained in <xref:S
   
 - For Transport Layer Security (TLS) best practices, see [Transport Layer Security (TLS) best practices with .NET Framework](tls.md).
 
-- Use <xref:System.Net.Http.HttpClient> to send HTTP requests instead of <xref:System.Net.WebRequest>, which was obsoleted in .NET 6. In .NET Framework, create a new `HttpClient` instance each time you need to send a request. (The guidance for .NET 5+/.NET Core is more nuanced. For more information, see [Guidelines for using HttpClient](../../fundamentals/networking/httpclient-guidelines.md).)
+- Use <xref:System.Net.Http.HttpClient> to send HTTP requests instead of <xref:System.Net.WebRequest>, which was obsoleted in .NET 6. In .NET Framework, create a new `HttpClient` instance each time you need to send a request. (The guidance for .NET 5+/.NET Core is more nuanced. For more information, see [Guidelines for using HttpClient](../../fundamentals/networking/http/httpclient-guidelines.md).)
```
[the remark this answers is not in the document]
Comment (IEvangelist) [reply docs/framework/network-programming/best-practices-for-system-net-classes.md:21]: Yes, it is. It's the past tense of _obsolete_.

File: docs/fundamentals/networking/http/http-overview.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+| `PATCH`   | ❌ No         | ❌ No                 | ❌ No   |
+| `DELETE`  | ✔️ Yes       | ❌ No                 | ❌ No   |
+| `HEAD`    | ✔️ Yes       | ✔️ Yes               | ✔️ Yes |
+| `OPTIONS` | ✔️ Yes       | ❌ No                 | ✔️ Yes |
+| `TRACE`   | ✔️ Yes       | ❌ No                 | ✔️ Yes |
+| `CONNECT` | ❌ No         | ❌ No                 | ❌ No   |
+
+> <sup>†</sup>The `POST` method is only cacheable when the appropriate `Cache-Control` or `Expires` response headers are present. This is very uncommon in practice.
+
+## HTTP status codes
```
Comment (BillWagner) [inline docs/fundamentals/networking/http/http-overview.md:44]: Should we link to the standard here? https://www.rfc-editor.org/rfc/rfc9110#name-status-codes

File: docs/fundamentals/networking/ipv6-overview.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    <settings>
+        <ipv6 enabled="true"/>
+        <!-- Omitted for brevity -->
+    </settings>
+    <!-- Omitted for brevity -->
+</system.net>
+```
+
+> [!NOTE]
+> IPv6 is enabled by default for .NET Framework version 2.0 and later. For .NET Framework version 1.1 and earlier, IPv6 is disabled by default.
```
Comment (BillWagner) [inline docs/fundamentals/networking/ipv6-overview.md:200]: This entire section appears to be .NET Framework specific. What is the corresponding information for .NET Core 1 - 3, and .NET 5 - 7?

**Review by BillWagner — COMMENTED** [review]
This is impressive @IEvangelist 

I found a few nits and questions while reading. Once we clarify those, this LGTM.

File: docs/fundamentals/networking/sockets/sockets-overview.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  - "receiving data, sockets"
+  - "protocols, sockets"
+  - "Internet, sockets"
+---
+
+# Sockets in .NET
+
+The <xref:System.Net.Sockets> namespace contains a managed, cross-platform socket networking implementation. All other network-access classes in the <xref:System.Net> namespace are built on top of this implementation of sockets.
+
+The .NET Framework <xref:System.Net.Sockets.Socket> class is a managed-code version of the socket services provided by the Winsock32 API. In most cases, the `Socket` class methods simply marshal data into their native Win32 counterparts and handle any necessary security checks.
```
[the remark this answers is not in the document]
Comment (IEvangelist) [reply docs/fundamentals/networking/sockets/sockets-overview.md:26]: Ah, yes - this is a mistake. I'll update this as it is now cross-platform.

File: docs/fundamentals/networking/http/http-overview.md (hunk 2 above)
Comment (IEvangelist) [reply docs/fundamentals/networking/http/http-overview.md:44]: I'll add a link.

File: docs/fundamentals/networking/ipv6-overview.md (hunk 3 above)
Comment (IEvangelist) [reply docs/fundamentals/networking/ipv6-overview.md]: This is configured with the env var. I'll update this section too.

File: docs/fundamentals/networking/sockets/sockets-overview.md [hunk 5]
```diff
@@ -23,7 +23,7 @@ helpviewer_keywords:
 
 The <xref:System.Net.Sockets> namespace contains a managed, cross-platform socket networking implementation. All other network-access classes in the <xref:System.Net> namespace are built on top of this implementation of sockets.
 
-The .NET Framework <xref:System.Net.Sockets.Socket> class is a managed-code version of the socket services provided by the Winsock32 API. In most cases, the `Socket` class methods simply marshal data into their native Win32 counterparts and handle any necessary security checks.
+The <xref:System.Net.Sockets.Socket> class is a managed-code version of the socket services provided relying on native interoperability with either Linux or Windows. In most cases, the `Socket` class methods simply marshal data into their native counterparts and handle any necessary security checks.
```
Comment (BillWagner) [inline docs/fundamentals/networking/sockets/sockets-overview.md:26]: We should call out MacOS specifically. (It's a unix OS, but it's not a Linux distro):

Comment (IEvangelist) [reply docs/fundamentals/networking/sockets/sockets-overview.md:26]: Great callout, I was a bit confused by that myself - so yes. The two classes are `Socket.Unix` and `Socket.Windows` in the source code.

Comment (BillWagner) [reply docs/fundamentals/networking/sockets/sockets-overview.md:26]: Thanks. I'll resolve and approve.

**Review by BillWagner — APPROVED** [review]
Thanks @IEvangelist 

This all LGTM. Let's :shipit:

File: docs/fundamentals/networking/sockets/sockets-overview.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  - "receiving data, sockets"
+  - "protocols, sockets"
+  - "Internet, sockets"
+---
+
+# Sockets in .NET
+
+The <xref:System.Net.Sockets> namespace contains a managed, cross-platform socket networking implementation. All other network-access classes in the <xref:System.Net> namespace are built on top of this implementation of sockets.
+
+The <xref:System.Net.Sockets.Socket> class is a managed-code version of the socket services provided relying on native interoperability with either Linux or Windows. In most cases, the `Socket` class methods simply marshal data into their native counterparts and handle any necessary security checks.
```
Comment (IEvangelist) [inline docs/fundamentals/networking/sockets/sockets-overview.md:26]:
```suggestion
The <xref:System.Net.Sockets.Socket> class is a managed-code version of the socket services provided relying on native interoperability with either Unix or Windows. In most cases, the `Socket` class methods simply marshal data into their native counterparts and handle any necessary security checks.
```

**Pull request merged**